### PR TITLE
chore: remove bar charts manual docs

### PR DESCRIPTION
**Pull request by denieler**
[FX-2568]

### Description

Remove manually entered docs and replace them with automatically generated docs, based on props.

| Before | After |
| ---- | ---- |
| Screenshot 2022-04-05 at 10 54 23 | Screenshot 2022-04-05 at 10 54 32 |
| ---- | ---- |
| Screenshot 2022-04-05 at 10 56 11 | Screenshot 2022-04-05 at 10 56 20 |
| ---- | ---- |
| Screenshot 2022-04-05 at 10 57 35 | Screenshot 2022-04-05 at 10 57 52 |

<details>

<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2568]: https://toptal-core.atlassian.net/browse/FX-2568?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso-charts/src/BarChart/BarChart.tsx [hunk 1]
```diff
@@ -29,10 +29,23 @@ const {
...
+    [key in K]: number
+  }
+}
+
 export interface Props<K extends string | number | symbol>
   extends BaseChartProps {
-  data: { name: string; value: { [key in K]: number } }[]
+  /**
+   * A list of data points to be rendered as a bar chart
+   * @type { name: string; value: { [key in K]: number }; }[]
```
Comment (dmaklygin) [inline packages/picasso-charts/src/BarChart/BarChart.tsx:43]: do you think, this information about type is necessary here? we've already declared it above.

File: .storybook/components/Markdown/Markdown.tsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from 'react'
 
 import ReactMarkdown from 'react-markdown'
+import remarkGfm from 'remark-gfm'
```
Comment (denieler) [inline .storybook/components/Markdown/Markdown.tsx:4]: after upgrading `react-markdown` this plugin is needed to support tables in md (from their docs)

File: package.json [hunk 3]
```diff
@@ -155,6 +156,7 @@
     "prismjs": "^1.25.0",
     "unset-value": "^2.0.1",
     "semver-regex": "^3.1.3",
-    "node-fetch": "^2.6.7"
+    "node-fetch": "^2.6.7",
+    "react-docgen-typescript": "^1.22.0"
```
Comment (denieler) [inline package.json:160]: we need to lock `react-docgen-typescript` version until we upgrade typescript to ^4.4 version cc @augustobmoura

File: .storybook/components/Markdown/Markdown.tsx (hunk 2 above)
Comment (dmaklygin) [reply .storybook/components/Markdown/Markdown.tsx:4]: okay, got it :)

**Review by dmaklygin — APPROVED** [review]

File: packages/picasso-charts/src/BarChart/BarChart.tsx (hunk 1 above)
Comment (denieler) [reply packages/picasso-charts/src/BarChart/BarChart.tsx:43]: @dmaklygin yes, automatically it doesn't show expanded object and shows just `BarChartDataItem<K>[]`, which is hard to use

Comment (denieler) [reply packages/picasso-charts/src/BarChart/BarChart.tsx:43]: @dmaklygin ah, yeah and `{ name: string; value: { [key in K]: number } }[]` it just can't recognize 🤷‍♂️ failing with the error. So to make automatic docgen to work I had to replace it with a type `BarChartDataItem`

**Review by OndrejTuma — APPROVED** [review]
Nice!

**Pull request merged**